src/distrs/wilcox.jl

```diff
@@ -0,0 +1,155 @@
+#=
+The wilcox distribution is equivalent to the problem
+of finding the number of subsets with size nx
+of {1,2,...,nx,nx+1,...,nx+ny} summing to (U-sum(1:nx))
+relative to the total number of subsets with size nx.
+The empty subset is defined to sum to zero.
+This can be calculated using the recursion:
+either nx+ny is in the subset in witch case we need to calculate
+the number of subsets with size nx-1 of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U-sum(1:nx)-nx-ny),
+or nx is not in the subset in which case we need to calculate
+the number of subsets with size nx of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U-sum(1:nx)),
+This can be calculated bottom up using dynamic programming.
+
+The (i,j)'th element of DP in the k'th outer loop iteration represents:
+the number of subsets with size k of {1,2,...,k,k+1,...,k+j-1} summing to i.
+ =#
+
+@inline function wilcoxDP(nx, ny, U)
+    DP = zeros(Int, sum(1:nx)+U, ny + 1)
+    for i in 1:(nx+ny)
+        for j in i:(ny+1)
+            DP[i, j] = 1
+        end
+    end
+    for k in 2:nx
+        for j in 1:(ny+1)
+            for i in min(U + sum(1:k), sum((j):(k+j-1))):-1:1
+                if j > 1
+                    DP[i, j] = DP[i, j-1]
+                else
+                    DP[i, j] = 0
+                end
+                if i - j - k + 1 >= 1
```

Comment:
This is just
```suggestion
                if i >= j + k
```
As above, maybe just split iteration over `i` in two parts, `for i in (a + min(U, b)):-1:(j + k)` and `for i in (j + k - 1):-1:1`?